docs/LibraryEvolution.rst

```diff
@@ -840,9 +853,18 @@ There are very few safe changes to make to protocols and their members:
   be added to a function requirement without any additional versioning
   information.
 
+Note that clients can use a protocol as an existential type as long as it does
+not have any associated types or requirements containing ``Self`` in
+non-contravariant position. Therefore adding such a requirement becomes a
+`binary-compatible source-breaking change`.
+
+Adding such requirements to protocols that already contain such requirements and
+cannot be used as existential types, on the other hand, is always
+source-compatible.
```

Comment:
This is the thing I don't want to promise, because of fully-constrained sub-protocols.